tcms/testplans/static/testplans/js/get.js

```diff
@@ -0,0 +1,145 @@
+        if (!testPlanId) {
+            return;
+        }
+
+        getTestCases();
+    };
+
+    var getTestCases = function() {
+        jsonRPC('TestCase.filter', [{'plan': testPlanId}], function(data) {
+            for (var i = 0; i < data.length; i++) {
+                var testCase = data[i];
+                if (testCase.case_status_id === 2) {
+                    testCases.confirmed.push(data[i]);
+                } else {
+                    testCases.not_confirmed.push(data[i]);
+                }
+            }
+            drawConfirmedTestCases();
+            addTestCasesAction();
+
+        });
+    };
+
+    var drawConfirmedTestCases = function() {
+        var mainTestCaseContainer = document.getElementById('confirmed-testcases'),
+            testCaseListHtml = '';
+
+        if (!mainTestCaseContainer) {
+            return;
+        }
+
+        if (testCases.confirmed.length > 0) {
+            for (var i = 0; i < testCases.confirmed.length; i++) {
+                testCaseListHtml += generateConfirmedCaseItem(testCases.confirmed[i]);
+            }
+        } else {
+            testCaseListHtml = `<div class="list-group-item">
+                                                <div class="list-group-item-header">
+                                                    <div class="list-view-pf-main-info">No testcases</div>
+                                                </div>
+                                            </div>`;
+        }
+        mainTestCaseContainer.innerHTML = testCaseListHtml;
+    };
+
+    var generateConfirmedCaseItem = function(testCase) {
+        return `<div class="list-group-item">
+                    <div class="list-group-item-header">
+                        <div class="list-view-pf-expand">
+                            <span class="fa fa-angle-right"></span>
+                        </div>
+                        <div class="list-view-pf-checkbox">
+                            <input type="checkbox"></div>
+                        <div class="list-view-pf-actions">
+                            <div class="dropdown pull-right dropdown-kebab-pf">
+                                <button class="btn btn-link dropdown-toggle" type="button" id="dropdownKebabRight9"
+                                        data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
+                                    <span class="fa fa-ellipsis-v"></span>
+                                </button>
+                                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="dropdownKebabRight9">
+                                    <li><a href="#">Action</a></li>
+                                    <li><a href="#">Another Action</a></li>
+                                    <li><a href="#">Something Else Here</a></li>
+                                    <li role="separator" class="divider"></li>
+                                    <li><a href="#">Separated Link</a></li>
```

Comment:
From what I can tell the only actions we need here is changing the TC status to something else (but that may change). 

Have a look at the `<template>` HTML tag:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template

Not ideal for every situation (from what I can tell you don't have place holders) but will allow defining the entire HTML structure in the template where you can also loop over context variables for the statuses.

Reply:
Its not supported by IE.

Reply:
We definitely don't care about IE. They can use Firefox instead.